frontend/src/hooks/useProducts.ts

```diff
@@ -77,16 +77,19 @@ export function useProducts() {
   const [products, setProducts] = useState([] as Product[])
 
   const sendQuery = useCallback(
-    (query: string) => {
+    (query: string, setLoading: (loading: boolean) => void) => {
       if (!query) {
         //TODO add error handling
         return
       }
+      
+      setLoading(true)
 
       // Fake API call
       setTimeout(() => {
         setProducts(testProducts)
         alert(`Query Received: ${query}`)
+        setLoading(false)
       }, 1000)
```

Comment:
`setLoading` should not be an input to the `sendQuery` function! Perhaps it is more appropriate to shift the `loading` and `setLoading` states to the context? (Instead of the `App` component)

Reply:
Ok this makes sense!  I was wondering where to handle the state changes too. 

I have attempted to shift the `loading` and `setLoading` states by implementing useLoading() using `useContext`. I wrap the context around the whole app so that we can use the loading spinner whenever we need it within the app.

Additionally, `setLoading` has been removed as an input to the `sendQuery` function. Please see commit [22da40b](https://github.com/nknguyenhc/luminous-zephyr/pull/16/commits/22da40b7407cdec01ac9a4f0aef421d28bec604d) to see if this is the intended code quality.